inst/pages/integrated_learner.qmd

```diff
@@ -19,518 +17,161 @@ advanced method, called cooperative learning [@Ding2022], which is also known as
-functional (pathway abundances) profiles.
-
-```{r load-pkg-data}
-
-##################
-# Load iHMP data #
-##################
-
-library(curatedMetagenomicData)
-library(dplyr)
-library(tidyverse)
-
-se_relative <- sampleMetadata |>
-  filter(study_name == "HMP_2019_ibdmdb") |>
-  returnSamples("relative_abundance", rownames = "short")
-
-se_pathway <- sampleMetadata |>
-  filter(study_name == "HMP_2019_ibdmdb") |>
-  returnSamples("pathway_abundance", rownames = "short")
-
-```
-
-We will first prepare the input sample metadata and feature table for both
-relative abundance and pathway abundance data.
-
-```{r prepare-metadata-feature}
+## Prediction of fat and prebiotics in diet of rats
 
-##########################
-# Create sample metadata #
-##########################
+As an example we would build a random forest classifier based on the 
+HintikkaxOData multi-omics profiles to classify fat and prebiotics in diet of rats. 
 
-sample_metadata <- select(
-    as.data.frame(colData(se_relative)),
-    c("study_name", "disease", "subject_id"))
+### With validation set
 
-# Define response variable & sample id
-sample_metadata$Y <- ifelse(sample_metadata$disease == "IBD", 1, 0)
-sample_metadata <- sample_metadata %>%
-    select(subject_id, Y) %>% rename(subjectID = subject_id)
+```{r valid-set}
 
 ###########################
-# Create Species Features #
-###########################
-
-feature_species <- as.data.frame(assay(se_relative))
-rownames(feature_species) <- sub('.*s__', '', rownames(feature_species))
-
+# load required libraries #
 ###########################
-# Create Pathway Features #
-###########################
-
-feature_pwys <- as.data.frame(assay(se_pathway))
-feature_pwys <- rownames_to_column(feature_pwys, "ID")
-feature_pwys <- feature_pwys %>%
-  filter(!grepl("\\|", ID)) %>%
-  filter(!ID %in% c('UNMAPPED', 'UNINTEGRATED')) %>%
-  column_to_rownames('ID') %>%
-  as.data.frame()
-
-################################
-# Combine Species and Pathways #
-################################
-
-feature_table <- bind_rows(feature_species, feature_pwys)
-
-########################################################
-# Check row names of feature_table and sample_metadata #
-########################################################
-
-all(colnames(feature_table) == rownames(sample_metadata))
-
-```
-
-We will then create a metadata table for the features. This table captures
-high-level information related to the features (e.g., which layer they belong
-to).
+library(mia)
+library(SuperLearner)
+library(IntegratedLearner)
+library(tidyverse)
+library(cowplot)
 
-```{r create-metadata-table}
+########################
+# load multi-omic data #
+########################
+data("HintikkaXOData")
+mae <- HintikkaXOData
 
 ######################################
-# Create metadata table for features #
+# Add validation set info to colData #
 ######################################
```

Comment:
We could remove this kind of comments as the other chunks of book do not have these either. Use the normal style

Reply:
ACK.